Client/metrics.yaml

```diff
@@ -3767,3 +3767,89 @@ credit_card:
     notification_emails:
       - fx-ios-data-stewards@mozilla.com
     expires: "2023-12-01"
+  form_detected:
+    type: event
+    description: |
+      Recorded when the user taps on credit card form input
+      and we detect it.
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXIOS-6703
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/15251
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2023-12-31"
+  autofilled:
+    type: event
+    description: |
+      Recorded when a user taps a card from bottom sheet
+      select a card flow and the credit card does not
+      get autofilled on the webpage page
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXIOS-6703
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/15251
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2023-12-31"
+  autofill_failed:
+    type: event
+    description: |
+      Recorded when a user taps a card from bottom sheet
+      select a card flow and the credit card does not
+      get autofilled on the webpage page
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXIOS-6703
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/15251
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2023-12-31"
+  save_prompt_create:
+    type: event
+    description: |
+      Recorded when a user saved a credit card using
+      the autofill save prompt.
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXIOS-6703
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/15251
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2023-12-31"
+  autofill_enabled:
+    type: event
+    description: |
+      Recorded when a user has save and autofill cards
+      enabled in autofill settings
+    lifetime: application
+    extra_keys:
+      is_enabled:
+        type: string
+        description: |
+          If is enabled or not (true / false)
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXIOS-6703
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/15251
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2023-12-31"
+  sync_enabled:
+    type: event
+    description: |
+      Recorded when a user has save and autofill cards
+      enabled in sync settings
+    lifetime: application
```

Comment:
Ah, not for `event` metrics. An `event` metric is permission to record as many as you'd like. Having the same event be sent in multiple pings doesn't really make sense since the timestamp would lie outside of subsequent pings' measurement windows.  Events are, generally, just for things where you care about what order they happened in relative to other events.

Which is why I wonder...are you sure you want an `event` metric? It sounds as though you're less interested in _when_ the value changed as _what_ value it had when the ping was sent. In which case it'd be

```yaml
type: boolean
description: |
  Whether save and autofill cards are enabled in sync settings.
  Set at startup in [source code link](), and whenever the value changes.
lifetime: application
```

And if it's important to know what the value of this is to contextualize the events being recorded, you should be sure to put it on the same ping that the events are sent on (which is the builtin "events" ping):
```yaml
send_in_pings:
  - events
```

Admittedly, I'm very confused by the use of a wrapper. Glean's designed to be used directly: routing through a wrapper makes it hard for me to understand what exactly is going on, so my advice might be a little off.

You might benefit from asking Travis Long (our resident iOS person) when he returns from PTO on Tuesday.

Reply:
Some day we can sit and I can walk you through the use of wrappers vs not. For now I am making the change as we discussed. 

Appreciate the detailed answer @chutten 💛